src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/validators/PathValidator.java

```diff
@@ -0,0 +1,456 @@
+        }
+
+        return Collections.emptySet();
+    }
+
+    private Set<String> validateReverseLink(InputData inputData, RepositoryData repositoryData) {
+        if (!repositoryData.getSwitchRepository().findById(inputData.getSegment().getDestSwitchId()).isPresent()) {
+            return Collections.singleton(getDestSwitchNotFoundError(inputData));
+        }
+        if (!repositoryData.getSwitchRepository().findById(inputData.getSegment().getSrcSwitchId()).isPresent()) {
+            return Collections.singleton(getSrcSwitchNotFoundError(inputData));
+        }
+
+        Optional<Isl> reverse = repositoryData.getIslRepository().findByEndpoints(
+                inputData.getSegment().getDestSwitchId(),
+                inputData.getSegment().getDestPort(),
+                inputData.getSegment().getSrcSwitchId(),
+                inputData.getSegment().getSrcPort());
+
+        if (!reverse.isPresent()) {
+            return Collections.singleton(getNoReverseIslError(inputData));
+        }
+        if (reverse.get().getStatus() != IslStatus.ACTIVE) {
+            return Collections.singleton(getReverseIslNotActiveError(inputData));
+        }
+
+        return Collections.emptySet();
+    }
+
+    private Set<String> validateBandwidth(InputData inputData, RepositoryData repositoryData) {
+        Optional<Isl> forward = repositoryData.getIslRepository().findByEndpoints(
+                inputData.getSegment().getSrcSwitchId(), inputData.getSegment().getSrcPort(),
+                inputData.getSegment().getDestSwitchId(), inputData.getSegment().getDestPort());
+        if (!forward.isPresent()) {
+            return Collections.singleton(getNoForwardIslError(inputData));
+        }
+
+        Set<String> errors = new HashSet<>();
+        if (getForwardBandwidthWithReusableResources(forward.get(), inputData) < inputData.getPath().getBandwidth()) {
+            errors.add(getForwardBandwidthErrorMessage(inputData, forward.get().getAvailableBandwidth()));
+        }
+
+        Optional<Isl> reverse = repositoryData.getIslRepository().findByEndpoints(
+                inputData.getSegment().getDestSwitchId(), inputData.getSegment().getDestPort(),
+                inputData.getSegment().getSrcSwitchId(), inputData.getSegment().getSrcPort());
+        if (!reverse.isPresent()) {
+            return Collections.singleton(getNoForwardIslError(inputData));
+        }
+        if (getReverseBandwidthWithReusableResources(reverse.get(), inputData) < inputData.getPath().getBandwidth()) {
+            errors.add(getReverseBandwidthErrorMessage(inputData, reverse.get().getAvailableBandwidth()));
+        }
+
+        return errors;
+    }
+
+    private long getForwardBandwidthWithReusableResources(Isl isl, InputData inputData) {
+        return getBandwidthWithReusableResources(inputData, isl,
+                pathSegment -> inputData.getSegment().getSrcSwitchId().equals(pathSegment.getSrcSwitchId())
```

Comment:
Four lines lambda is too hard to read. Please, move to the method with self-explaining name, if possible.
(same for below methods)

Reply:
fixed